### PR TITLE
Spiders: Prevents player-controlled spiders from reproducing without a directive. 

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request

Completely prevents player controlled spiders from reproducing with a null directive and also prevents accidentally clearing directive

## Why It's Good For The Game

Spiders without directives cause mass mayhem and chaos, often times without a clear path of blame. Without a way to return to a null directive, bad directives can always be traced back to someone intentionally setting a new one. 

## Changelog
:cl:
tweak: Reproductive spiders may no longer accidentally clear their directives, and may not reproduce without one set. 
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
Working on preventing reproduction entirely by player controlled spiders without a directive.

**Comment by Rukofamicom** [comment]
Now I'd just like to fix error messages when failing to lay eggs, but by all rights this is good to go. 

*The error messages for why you failed to lay eggs already didn't work*

**Comment by Rukofamicom** [comment]
No error message fix, but I did update the ability icon so that players are at least told they need to set a directive in order to reproduce on the reproduction ability itself.

**Comment by Archanial** [comment]
Ided pr

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 1]
```diff
@@ -406,15 +406,17 @@
...
 	button_icon_state = "lay_eggs"
 
 /datum/action/innate/spider/lay_eggs/IsAvailable()
 	if(..())
 		if(!istype(owner, /mob/living/simple_animal/hostile/poison/giant_spider/nurse))
 			return 0
 		var/mob/living/simple_animal/hostile/poison/giant_spider/nurse/S = owner
-		if(S.fed)
+		if(S.fed && S.directive)
+			return 1
```
Comment (Archanial) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]:
```suggestion
			return TRUE
```
If you can change rest returns to true/false

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 2]
```diff
@@ -406,15 +406,17 @@
...
+	desc = "Lay a cluster of eggs, which will soon grow into more spiders. You must have a directive set and wrap a living being to do this."
 	button_icon_state = "lay_eggs"
 
 /datum/action/innate/spider/lay_eggs/IsAvailable()
 	if(..())
 		if(!istype(owner, /mob/living/simple_animal/hostile/poison/giant_spider/nurse))
 			return 0
 		var/mob/living/simple_animal/hostile/poison/giant_spider/nurse/S = owner
-		if(S.fed)
+		if(S.fed && S.directive)
```
Comment (Archanial) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: You are checking for fed twice, what you should do is check fed then return true whenever player has no ckey or has directive.

**Review by Archanial — CHANGES_REQUESTED** [review]

**Review by Archanial — APPROVED** [review]
Code good

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**